WordPress/Classes/ViewRelated/Reader/Select Interests/ReaderInterestsCollectionViewFlowLayout.swift

```diff
@@ -0,0 +1,198 @@
+            if attribute.frame.intersects(rect) {
+                layoutAttributes.append(attribute)
+            }
+        }
+
+        return layoutAttributes
+    }
+
+    private func centeredLayoutAttributesForElements(in rect: CGRect) -> [UICollectionViewLayoutAttributes]? {
+        var rows = [Row]()
+        var rowY: CGFloat = CGFloat.greatestFiniteMagnitude
+
+        // Create an array of "rows" based on the y positions
+        for attribute in layoutAttributes {
+            if attribute.frame.intersects(rect) == false {
+                continue
+            }
+
+            let minY = attribute.frame.minY
+
+            if rowY != minY {
+                rowY = minY
+
+                rows.append(Row(itemSpacing: itemSpacing, isRightToLeft: isRightToLeft))
+            }
+
+            rows.last?.add(attribute: attribute)
+        }
+
+        return rows.flatMap { (row: Row) -> [UICollectionViewLayoutAttributes] in
+            return row.centeredLayoutAttributesIn(width: maxContentWidth)
+        }
+    }
+
+    override open func layoutAttributesForItem(at indexPath: IndexPath) -> UICollectionViewLayoutAttributes? {
+        return layoutAttributes[indexPath.row]
+    }
+}
+
+// MARK: - Row Centering Helper Class
+private class Row {
+    var layoutAttributes = [UICollectionViewLayoutAttributes]()
+    let itemSpacing: CGFloat
+    let isRightToLeft: Bool
+
+    init(itemSpacing: CGFloat, isRightToLeft: Bool) {
+        self.itemSpacing = itemSpacing
+        self.isRightToLeft = isRightToLeft
+    }
+
+    /// Add a new attribute to the row
+    /// - Parameter attribute: layout attribute to be added
+    public func add(attribute: UICollectionViewLayoutAttributes) {
+        layoutAttributes.append(attribute)
+    }
+
+    /// Calculates a new X position for each item in this row based on a new x offset
+    /// - Parameter width: The total width of the container view to center in
+    /// - Returns: The new centered layout attributes
+    public func centeredLayoutAttributesIn(width: CGFloat) -> [UICollectionViewLayoutAttributes] {
+        let centerX = (width - rowWidth) * 0.5
+
+        var offset = isRightToLeft ? width - centerX : centerX
+
+        for attribute in layoutAttributes {
+            let itemWidth = attribute.frame.width + itemSpacing
+
+            if isRightToLeft {
+                attribute.frame.origin.x = offset - attribute.frame.width
+                offset -= itemWidth
+            } else {
+                attribute.frame.origin.x = offset
+                offset += itemWidth
+            }
+        }
+
+        return layoutAttributes
+    }
+
+    /// Calculate the total row width including spacing
+    private var rowWidth: CGFloat {
+        let width = layoutAttributes.reduce(0, { width, attribute -> CGFloat in
+            return width + attribute.frame.width
+        })
+
+        return width + itemSpacing * CGFloat(layoutAttributes.count-1)
```

Comment:
Again super nitpicky: Can you please dd spaces before and after "-"
so : CGFloat(layoutAttributes.count - 1)